### PR TITLE
adds slug to users show

### PR DESCRIPTION
**Pull request by alfosco**
Adds slug to users show url; route shows up as '/your_username'
-amends route for users#show
-adds method to User model

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<h1>Account Info: <%= current_user.full_name %></h1>
+<h1>Account Info: <%= @user.first_name %> <%= @user.last_name %></h1>
```
Comment (cmacaulay) [inline app/views/users/show.html.erb]: @alfosco was the full_name method not working for you?

File: config/routes.rb [hunk 2]
```diff
@@ -17,7 +17,8 @@
   put '/password/update', to: 'passwords#update', as: "update_password"
 
   # users
-  resources :users, only: [:new, :create, :edit, :update, :show]
+  resources :users, path: '', only: [:show]
```
Comment (cmacaulay) [inline config/routes.rb]: should we put an alias in this route? @alfosco @AELSchauer @JonathanSR

File: spec/features/user/user_url_is_slugged_with_username_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    )
+
+    controller = ApplicationController
+    allow_any_instance_of(controller).to receive(:current_user).and_return(user)
+
+    visit user_path(user)
+
+    expect(current_path).to eq("/espinosa1")
+  end
+end
```
Comment (cmacaulay) [inline spec/features/user/user_url_is_slugged_with_username_spec.rb]: 👏 🎉 testing!

File: app/models/user.rb [hunk 4]
```diff
@@ -34,5 +34,8 @@ def registered_user
     self.roles << role
   end
 
+  def to_param
+    username.parameterize
+  end
```
Comment (cmacaulay) [inline app/models/user.rb]: 🎉

**Comment by cmacaulay** [comment]
@alfosco once you pull down from master and all tests pass, feel free to merge!

**Comment by cmacaulay** [comment]
@alfosco can you fix the merge conflicts and merge this to master?

File: spec/features/user/user_url_is_slugged_with_username_spec.rb (hunk 3 above)
Comment (JonathanSR) [inline spec/features/user/user_url_is_slugged_with_username_spec.rb]: TDD all the way!

File: app/views/shared/_upload_form.html.erb [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <%= form_for @file do |f| %>
-   <%= f.file_field :attachment %>
-   <%= f.text_field :folder_id, value: "#{current_folder.id}", type: :hidden %>
-   <%= f.submit "Upload File" %>
- <% end %>
+  <%= f.file_field :attachment %>
+  <%= f.text_field :folder_id, value: "#{current_folder.id}", type: :hidden %>
```
Comment (AELSchauer) [inline app/views/shared/_upload_form.html.erb]: We may be able to refactor and remove these from our forms entirely now that current_folder is tracked in the session.

File: spec/features/user/user_url_is_slugged_with_username_spec.rb (hunk 3 above)
Comment (AELSchauer) [inline spec/features/user/user_url_is_slugged_with_username_spec.rb]: 👍

**Pull request merged**